casper/src/test/scala/coop/rchain/casper/HashSetCasperTest.scala

```diff
@@ -523,6 +536,79 @@ class HashSetCasperTest extends FlatSpec with Matchers {
     node.tearDown()
   }
 
+  it should "allow paying for deploys" in {
+    val node      = HashSetCasperTestNode.standalone(genesis, validatorKeys.head)
+    val (sk, pk)  = Ed25519.newKeyPair
+    val user      = ByteString.copyFrom(pk)
+    val timestamp = System.currentTimeMillis()
+    val phloPrice = 1L
+    val amount    = 847L
+    val sigDeployData = ProtoUtil
+      .sourceDeploy(
+        s"""new retCh in { @"blake2b256Hash"!([0, $amount, *retCh].toByteArray(), "__SCALA__") }""",
+        timestamp,
+        accounting.MAX_VALUE
+      )
+      .withUser(user)
+    val sigData = node.runtimeManager
+      .captureResults(
+        genesis.getBody.getPostState.tuplespace,
+        ProtoUtil.deployDataToDeploy(sigDeployData)
+      )
+      .head
+      .exprs
+      .head
+      .getGByteArray
+    val sig   = Base16.encode(Ed25519.sign(sigData.toByteArray, sk))
+    val pkStr = Base16.encode(pk)
+    val paymentCode =
+      s"""new paymentForward, walletCh in {
+         |  for(faucet <- @"faucet"; pos <- @"proofOfStake"){
+         |    faucet!($amount, "ed25519", "$pkStr", *walletCh) |
+         |    for(@[wallet] <- walletCh) {
+         |      for(@purse <- paymentForward){ pos!("pay", purse, Nil) } |
+         |      @wallet!("transfer", $amount, 0, "$sig", *paymentForward, Nil)
```

Comment:
How does the verification here work? the `paymentForward` channel is the same as the one signed at the beginning because we use the same timestamp and user right?

Reply:
Yes